ui/tests/unit/lib/console-helpers-test.js

```diff
@@ -0,0 +1,328 @@
+    expectedData: {
+      type: 'json',
+      content: { foo: 'bar', wrap_info: { one: 'two' } },
+    },
+  },
+  {
+    name: 'with -format=json and -field flags',
+    args: [{ foo: 'bar', data: { one: 'two' } }, 'foo/bar', 'read', { format: 'json', field: 'one' }],
+    expectedData: {
+      type: 'json',
+      content: 'two',
+    },
+  },
+  {
+    name: 'with -format=json and -field, and -wrap-ttl flags',
+    args: [
+      { foo: 'bar', wrap_info: { one: 'two' } },
+      'foo/bar',
+      'read',
+      { format: 'json', wrapTTL: '1h', field: 'one' },
+    ],
+    expectedData: {
+      type: 'json',
+      content: 'two',
+    },
+  },
+  {
+    name: 'with string field flag and wrap-ttl flag',
+    args: [{ foo: 'bar', wrap_info: { one: 'two' } }, 'foo/bar', 'read', { field: 'one', wrapTTL: '1h' }],
+    expectedData: {
+      type: 'text',
+      content: 'two',
+    },
+  },
+  {
+    name: 'with object field flag and wrap-ttl flag',
+    args: [
+      { foo: 'bar', wrap_info: { one: { two: 'three' } } },
+      'foo/bar',
+      'read',
+      { field: 'one', wrapTTL: '1h' },
+    ],
+    expectedData: {
+      type: 'object',
+      content: { two: 'three' },
+    },
+  },
+  {
+    name: 'with response data and string field flag',
+    args: [{ foo: 'bar', data: { one: 'two' } }, 'foo/bar', 'read', { field: 'one', wrapTTL: '1h' }],
+    expectedData: {
+      type: 'text',
+      content: 'two',
+    },
+  },
+  {
+    name: 'with response data and object field flag ',
+    args: [
+      { foo: 'bar', data: { one: { two: 'three' } } },
+      'foo/bar',
+      'read',
+      { field: 'one', wrapTTL: '1h' },
+    ],
+    expectedData: {
+      type: 'object',
+      content: { two: 'three' },
+    },
+  },
+  {
+    name: 'response with data',
+    args: [{ foo: 'bar', data: { one: 'two' } }, 'foo/bar', 'read', {}],
+    expectedData: {
+      type: 'object',
+      content: { one: 'two' },
+    },
+  },
+  {
+    name: 'with response data, field flag, and field missing',
+    args: [{ foo: 'bar', data: { one: 'two' } }, 'foo/bar', 'read', { field: 'foo' }],
+    expectedData: {
+      type: 'error',
+      content: 'Field "foo" not present in secret',
+    },
+  },
+  {
+    name: 'with response data and auth block',
+    args: [{ data: { one: 'two' }, auth: { three: 'four' } }, 'auth/token/create', 'write', {}],
+    expectedData: {
+      type: 'object',
+      content: { three: 'four' },
+    },
+  },
+  {
+    name: 'with -field and -format with an object field',
+    args: [{ data: { one: { three: 'two' } } }, 'sys/mounts', 'read', { field: 'one', format: 'json' }],
+    expectedData: {
+      type: 'json',
+      content: { three: 'two' },
+    },
+  },
+  {
+    name: 'with -field and -format with a string field',
+    args: [{ data: { one: 'two' } }, 'sys/mounts', 'read', { field: 'one', format: 'json' }],
+    expectedData: {
+      type: 'json',
+      content: 'two',
+    },
+  },
+];
```

Comment:
Nice to see you are doing this fixture looping thing, something I really like 👍 , but I'd seen contradictory discussion elsewhere. I suppose its a case of sticking to what's in the project. Only thing I'd say is if they get a bit big like here, maybe have a think about how you could 'get them out of the way' and if you'd want to do that.